### ISSUE TITLE
refactor(backend): "Case" entity type should be abstract

### ISSUE THREAD

**Issue by labo-flg** [LuatixHQ/opencti#6684]
Container is an abstract type.
Case is a Container.
...and for instance, RFI is a Case.

Cases cannot be created as-is, they are not meant to. They are seemingly abstract.
However, they are registered like concrete entity types, with stix-converter and the like.

We need to :
* assess that Case are indeed abstract and should not be treated like concret types
* if confirmed, cleanup the case module accordingly

Seen while working on https://github.com/OpenCTI-Platform/opencti/pull/6672

**Comment by jborozco** [comment]
Hi @labo-flg, this looks more like a feature technical refinement than a bug to me. Do you agree to push it to the feature backlog ?

**Comment by labo-flg** [comment]
@jborozco you're right! This is definitely a technical debt issue, not causing any bug for now. It's really a matter of cleaning up this part of the code.
